stdlib/LinearAlgebra/test/lu.jl

```diff
@@ -72,6 +72,11 @@ dimg  = randn(n)/2
             bft = eltya <: Real ? LinearAlgebra.LU{BigFloat} : LinearAlgebra.LU{Complex{BigFloat}}
             bflua = convert(bft, lua)
             @test bflua.L*bflua.U ≈ big.(a)[p,:] rtol=ε
+            @test Factorization{eltya}(lua) == lua
```

Comment:
If this is meant to be a no-op in the case of matching eltypes, maybe strengthen this to `===` comparison?